AG-16097 - Cleanup new SonarCloud issues pt7

alantreadway opened the pull request:
https://ag-grid.atlassian.net/browse/AG-16097

alantreadway committed:
AG-16097 - Tighten ESLint rules that were recently eliminated.

alantreadway committed:
AG-16097 - Tighten ESLint rules that were recently eliminated.

lsjroberts reviewed (approved):
LGTM

iMoses reviewed (changes requested):
Don't change `main-modules.ts`
Add this: `/* eslint-disable unicorn/prefer-export-from */`

IMO this rule make sense generally, but not in the case of an entry file.

alantreadway committed:
AG-16097 - Review fixes.

alantreadway committed:
AG-16097 - Fix ESLint violations from newly enabled rules

Fixes violations from recently enabled ESLint rules:
- sonarjs/redundant-type-aliases: Added eslint-disable comments for intentional type aliases
- Remove unused unicorn/prefer-export-from eslint-disable comments

Changes:
- callbacks.test.ts: Disable redundant-type-aliases for intentional test fixture type
- dataModel.ts: Disable redundant-type-aliases for ScopeId type alias
- main-modules.ts: Remove unused eslint-disable comment (code follows proper export pattern)

alantreadway committed:
Merge remote-tracking branch 'origin/latest' into ag-16097/cleanup-new-sonarcloud-issues-pt6

alantreadway committed:
AG-16097 - Fix more eslint violations.

alantreadway commented:
> Don't change `main-modules.ts` Add this: `/* eslint-disable unicorn/prefer-export-from */`

Fixed

iMoses reviewed (approved):
LGTM

alantreadway committed:
AG-16097 - Fix more eslint violations.

alantreadway committed:
AG-16097 - Fix more eslint violations.
